### PR TITLE
added authentication filter and find by login methods

### PR DESCRIPTION
**Pull request by Martynice**

### CODE REVIEW

File: src/main/java/com/internet/shop/web/filter/AuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+@WebFilter("/*")
+public class AuthenticationFilter implements Filter {
+    private static final String USER_ID = "user_id";
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
+    }
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/web/filter/AuthenticationFilter.java]:
```suggestion
    public void init(FilterConfig filterConfig) throws ServletException {
    }
```

File: src/main/java/com/internet/shop/web/filter/AuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            resp.sendRedirect(req.getContextPath() + "/login");
+            return;
+        }
+        filterChain.doFilter(req, resp);
+    }
+
+    @Override
+    public void destroy() {
+
+    }
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/web/filter/AuthenticationFilter.java]:
```suggestion
    public void destroy() {
    }
```

**Review by dariiabutorina — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/controllers/user/LoginController.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String pwd = req.getParameter("pwd");
+        try {
+            User user = authService.login(login, pwd);
+            HttpSession session = req.getSession();
+            session.setAttribute("user_id", user.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/user/LoginController.java]: let's make `user_id` a constant

**Review by KseniiaMakarova — APPROVED** [review]

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
fix travis please

File: src/main/java/com/internet/shop/controllers/user/LoginController.java (hunk 3 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/controllers/user/LoginController.java]: done

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**